src/en/manga18me/src/eu/kanade/tachiyomi/extension/en/manga18me/Manga18Me.kt

```diff
@@ -0,0 +1,219 @@
+
+    override val lang = "en"
+
+    override val baseUrl = "https://manga18.me"
+
+    override val supportsLatest = true
+
+    override val client = network.client.newBuilder()
+        .rateLimit(3, 1)
+        .build()
+
+    override fun headersBuilder() = Headers.Builder().apply {
+        add("Referer", "$baseUrl/")
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/manga/$page?orderby=trending", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val entries = document
+            .select(popularMangaSelector())
+            .filter { it ->
+                val searchText = document.selectFirst("div.section-heading h1")?.text() ?: ""
+                val raw = document.selectFirst("div.canonical")?.attr("href") ?: ""
+                val title = it.selectFirst("div.item-thumb.wleft a")?.attr("href") ?: ""
+
+                if (searchText.lowercase().contains("raw")) {
+                    true
+                } else if (raw.lowercase().contains("raw")) {
+                    true
+                } else if (title.contains("raw")) {
+                    false
+                } else {
+                    true
+                }
+            }
+            .map(::popularMangaFromElement)
+        val hasNextPage = document.selectFirst(popularMangaNextPageSelector()) != null
+
+        return MangasPage(entries, hasNextPage)
+    }
+
+    override fun popularMangaSelector() = "div.page-item-detail"
+    override fun popularMangaNextPageSelector() = ".next"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        title = element.selectFirst("div.item-thumb.wleft a")!!.attr("title")
+        thumbnail_url = element.selectFirst("img")?.absUrl("src")
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/manga/$page?orderby=latest", headers)
+    }
+
+    override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
+    override fun latestUpdatesSelector() = popularMangaSelector()
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    /** Manga Search **/
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        return if (query.isEmpty()) {
+            client.newCall(searchMangaRequest(page, query, filters))
+                .asObservableSuccess()
+                .map(::searchMangaParse)
+        } else {
+            client.newCall(GET("$baseUrl/search?q=$query", headers))
+                .asObservableSuccess()
+                .map(::searchMangaParse)
+        }
+    }

```

Comment:
no need to override the fetch method, just add condition to `searchMangaRequest`

also, use `HttpUrl.Builder` to ensure query is properly encoded